src/discord_bot/bot.py

```diff
@@ -1,13 +1,21 @@
-import os
 import interactions
-import dotenv
+from pydantic_settings import BaseSettings, SettingsConfigDict
 from discord_bot.llm import answer_question
 
-dotenv.load_dotenv()
 
+class Settings(BaseSettings):
+  model_config = SettingsConfigDict(env_file="../../.env", env_file_encoding="utf-8")
+
+  DISCORD_BOT_TOKEN: str
+  AI_SERVER_URL: str = "http://localhost:8000"
+
+
+# mypy: ignore-errors
+# MyPy requires unecessary named arguments which are automatically imported by pydantic_settings.BaseSettings
```

Comment:
This isn't needed. This can be fixed by loading the pydantic mypy plugin in `pyproject.toml`

```toml
[tool.mypy]
# other things
plugins = [
  "pydantic.mypy"
]
```

Reply:
but yeah, even though with this plugin, pylance will still report an error. Hopefully this will be fixed in a future fix between mypy/pydantic/both.

Reply:
I do what you advised and it doesn't complain about it anymore